pybamm/parameters/lithium_ion_parameters.py

```diff
@@ -628,22 +695,34 @@ def _set_dimensionless_parameters(self):
         self.T_init = self.therm.T_init
         self.c_e_init = self.c_e_init_dimensional / self.c_e_typ
 
-    def chi(self, c_e):
+        # Dimensionless mechanical parameters
+        if isinstance(self, pybamm.particle_cracking.BaseCracking):
+            self.rho_cr_n = self.rho_cr_n_dim * self.l_cr_n_0 * self.w_cr
+            self.rho_cr_p = self.rho_cr_p_dim * self.l_cr_p_0 * self.w_cr
+            self.theta_p = self.theta_p_dim * self.c_p_max / self.T_ref
+            self.theta_n = self.theta_n_dim * self.c_n_max / self.T_ref
+            self.t0_cr = 3600 / self.timescale  # nomarlised typical time for one cycle
+
+    def chi(self, c_e, T):
         """
         Thermodynamic factor:
             (1-2*t_plus) is for Nernst-Planck,
             2*(1-t_plus) for Stefan-Maxwell,
         see Bizeray et al (2016) "Resolving a discrepancy ...".
         """
-        return (2 * (1 - self.t_plus(c_e))) * (self.one_plus_dlnf_dlnc(c_e))
+        T_dim = self.Delta_T * T + self.T_ref
+        return (2 * (1 - self.t_plus(c_e))) * (self.one_plus_dlnf_dlnc(c_e, T_dim))
```

Comment:
I think you should just pass `T` to `one_plus_dlnf_dlnc` here. you do the conversion from `T` to `T_dim` inside the definition of `one_plus_dlnf_dlnc`

Reply:
Not sure for this. In the parameter set, many functions are defined with dimentional `T_dim` for the arrhenius, and it will be confusing if mixed definition is used.

Reply:
yeah but you scale again inside one_plus_dlnf_dlnc so it is getting scaled twice?

Reply:
Okay, It was misleading with dimentioal `T_dim` in `chi`. I will make `T` dimentionless in `chi` and dimentional in `one_plus_dlnf_dlnc` instead.

Reply:
since `chi` is the thing called in the model, which is in dimensionless form, it should be a function of dimensionless `c_e` and `T`. similarly `one_plus_dlnf_dlnc` should be a function of dimensionless `c_e` and `T`, but then the `FunctionParameter` "1 + dlnf/dlnc" should get dimensional concentration and temperature as inputs, since this is the object that calls the function from `inputs/parameters` which is in dimensional form

i.e.

```python3
    def chi(self, c_e, T):
        """
        Thermodynamic factor:
            (1-2*t_plus) is for Nernst-Planck,
            2*(1-t_plus) for Stefan-Maxwell,
        see Bizeray et al (2016) "Resolving a discrepancy ...".
        """
        return (2 * (1 - self.t_plus(c_e))) * (self.one_plus_dlnf_dlnc(c_e, ))
```
and 
```python3
    def one_plus_dlnf_dlnc(self, c_e, T):
        inputs = {
            "Electrolyte concentration [mol.m-3]": c_e * self.c_e_typ,
            "Temperature [K]": T * self.Delta_T + self.T_ref,
        }
        return pybamm.FunctionParameter("1 + dlnf/dlnc", inputs)
```

I agree we should make this more clear though. Functions defined in "2. Dimensional Functions" should take dimensional inputs, but things defined in "5. Dimensionless Functions" should take dimensionless inputs.